### PR TITLE
Typescript 2.1

### PR DESCRIPTION
**Pull request by nathanrobinson**
updated diff.ts line 63 to be TS 2.1 compatible

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,12 +35,22 @@ Now the value of `users` is:
...
+Will generate:
+
+    [ {op: 'test', from: '', path: '/itemCodes/1', value: "456"} ]
 
 # API
 
 `rfc6902` exports two methods, no default. So in ES6 syntax, that would be:
 
-    import {applyPatch, createPatch} from 'rfc6902';
+    import {applyPatch, createPatch, createTests} from 'rfc6902';
```
Comment (chbrown) [inline README.md]: It's not a typical use so dunno if I'd include this here, but if you do, change the line above to 'three'

File: index.ts [hunk 2]
```diff
@@ -46,3 +46,31 @@ export function createPatch(input, output): Operation[] {
...
+/**
+Produce an 'application/json-patch+json'-type patch to test a patch before
+working with existing values.
+
+This does not alter `input` or `output` unless they have a property getter with
+side-effects (which is not a good idea anyway).
+
+Returns list of test operations to perform on `input` to validate `patch`.
+*/
+export function createTests(input, patch: Operation[]): Operation[] {
```
Comment (chbrown) [inline index.ts]: I think TypeScript is happier with explicit any? I.e. `input: any`

File: index.ts [hunk 3]
```diff
@@ -46,3 +46,31 @@ export function createPatch(input, output): Operation[] {
...
+  let tests = new Array<Operation>();
```
Comment (chbrown) [inline index.ts]: I generally go with `const` wherever possible. Perhaps there are performance optimizations that prefer let, though?

File: index.ts [hunk 4]
```diff
@@ -46,3 +46,31 @@ export function createPatch(input, output): Operation[] {
...
+
+  let patchesToTest = patch.filter(p => p.op === 'remove'
+                                     || p.op === 'replace'
+                                     || p.op === 'copy'
+                                     || p.op === 'move');
+
+  for(let op of patchesToTest) {
+    const pointer = new Pointer(op.path.split('/'));
```
Comment (chbrown) [inline index.ts]: Prefer to use Pointer.fromJSON static method (handles unescaping and throwing on invalid syntax)

File: index.ts [hunk 5]
```diff
@@ -46,3 +46,31 @@ export function createPatch(input, output): Operation[] {
...
+    const pathItem = pointer.evaluate(input);
+    if(pathItem){
```
Comment (chbrown) [inline index.ts]: Errm, hate to be pedantic but please match code style in codebase (`if (pathItem) {` etc.)

**Comment by nathanrobinson** [comment]
git isn't letting me squash since I've already pushed, so I guess I'll have to cancel this PR and create a new branch.

**Pull request closed without merging**